service/history/timer_queue_task_executor_base.go

```diff
@@ -225,6 +225,11 @@ func (t *timerQueueTaskExecutorBase) executeSingleStateMachineTimer(
 	ref := hsm.Ref{
 		WorkflowKey:     ms.GetWorkflowKey(),
 		StateMachineRef: timer.Ref,
+		Validate:        smt.Validate,
+	}
+	// TODO(bergundy): I think we never want to execute tasks on zombie workflows, not sure why we check write access.
```

Comment:
I defined validateNotZombieWorkflow for the Access func, which will be used for Get api implementation and we still want to support read operation on zombie workflows/state machines.

Reply:
Got it. I'm pushing a commit that always checks that a workflow is not zombie for task references since I don't see a case where we want to allow accessing a machine in a zombie workflow in the task processing stack.